src/main/java/com/thomashan/coup/turn/state/WaitingToPerformActionsState.java

```diff
@@ -0,0 +1,148 @@
+package com.thomashan.coup.turn.state;
+
+import com.thomashan.coup.action.Action;
+import com.thomashan.coup.action.MainAction;
+import com.thomashan.coup.action.MainActionType;
+import com.thomashan.coup.action.NoAction;
+import com.thomashan.coup.action.PerformAction;
+import com.thomashan.coup.action.PerformActionType;
+import com.thomashan.coup.card.Deck;
+import com.thomashan.coup.card.DrawnCard;
+import com.thomashan.coup.player.Player;
+import com.thomashan.coup.player.Players;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+import static java.util.Optional.empty;
+
+public class WaitingToPerformActionsState implements TurnState<PerformAction> {
+    private final Players players;
+    private final Player player;
+    private final Deck deck;
+    private final MainAction mainAction;
+    private final List<Action> actionHistory;
+    private final Optional<Player> target;
+
+    private WaitingToPerformActionsState(Players players, Player player, Deck deck, MainAction mainAction, List<Action> actionHistory, Player target) {
+        this.players = players;
+        this.player = player;
+        this.deck = deck;
+        this.mainAction = mainAction;
+        this.actionHistory = actionHistory;
+
+        if (target != null) {
+            this.target = Optional.of(target);
+        } else {
+            this.target = empty();
+        }
+    }
+
+    static WaitingToPerformActionsState of(Players players, Player player, Deck deck, MainAction mainAction, List<Action> actionHistory, Player target) {
+        return new WaitingToPerformActionsState(players, player, deck, mainAction, actionHistory, target);
+    }
+
+    @Override
+    public List<Action> getActionHistory() {
+        return actionHistory;
+    }
+
+    @Override
+    public Deck getDeck() {
+        return deck;
+    }
+
+    @Override
+    public Players getPlayers() {
+        return players;
+    }
+
+    @Override
+    public Player getPlayer() {
+        return player;
+    }
+
+    @Override
+    public Optional<Player> getTarget() {
+        return target;
+    }
+
+    @Override
+    public MainAction getMainAction() {
+        return mainAction;
+    }
+
+    @Override
+    public boolean isComplete() {
+        return false;
+    }
+
+    @Override
+    public boolean isCheckAllowableActions() {
+        return false;
+    }
+
+    @Override
+    public List<PerformAction> getAllowableActions() {
+        return Collections.emptyList();
+    }
+
+    @Override
+    public TurnState performAction(PerformAction action) {
+        if (action.getActionType() instanceof MainActionType) {
+            MainActionType mainActionType = (MainActionType) action.getActionType();
+
+            switch (mainActionType) {
+                case ASSASSINATE: {
+                    Player newPlayer = getPlayer().useCoinsForAssassination();
+                    Players newPlayers = getPlayers().updatePlayer(getPlayer(), newPlayer);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Found 'DU'-anomaly for variable 'newPlayers' (lines '99'-'147').](https://app.codacy.com/app/thomashan/coup-java/pullRequest?prid=4561385)